### PR TITLE
Updated Implementation/Listeners

### PR DESCRIPTION
**Pull request by pva701**

### CODE REVIEW

File: _docs/for-contributors/2017-01-02-implementation.md [hunk 1]
```diff
@@ -59,25 +59,32 @@ block (if you're the selected stakeholder), repeat.
...
+Various supplemental listeners will not be covered, focusing on the main ones instead.
+
+Mostly listeners use *Relay* framework, which inludes three type of messages:
+  - `Inventory` message: node publishes message to network when gets a new data.
+  - `Request` message: node requests a new data which  was published in `Inventory` message.
+  (from other node), if this data is not known yet by this node
+  - `Data` message: node replies with this message on `Request` message. `Data` message contains concrete data.
+
+For instance, when user creates a new transaction, wallet sends `Inventory` message with
+transaction id to network, if node which recieved `Inventory` doesn't know transaction with such id,
```
Comment (rockbmb) [inline _docs/for-contributors/2017-01-02-implementation.md]: Typo @pva701, `received`

File: _docs/for-contributors/2017-01-02-implementation.md [hunk 2]
```diff
@@ -59,25 +59,32 @@ block (if you're the selected stakeholder), repeat.
...
+Mostly listeners use *Relay* framework, which inludes three type of messages:
+  - `Inventory` message: node publishes message to network when gets a new data.
+  - `Request` message: node requests a new data which  was published in `Inventory` message.
+  (from other node), if this data is not known yet by this node
+  - `Data` message: node replies with this message on `Request` message. `Data` message contains concrete data.
+
+For instance, when user creates a new transaction, wallet sends `Inventory` message with
+transaction id to network, if node which recieved `Inventory` doesn't know transaction with such id, 
+then it replies with `Request` message, after that wallet sends this transaction in `Data` message.
+After node recieved `Data` message it can send `Inventroy` message
```
Comment (rockbmb) [inline _docs/for-contributors/2017-01-02-implementation.md]: Same here. Also, `Inventroy` -> `Inventory`

File: _docs/for-contributors/2017-01-02-implementation.md [hunk 3]
```diff
@@ -59,25 +59,32 @@ block (if you're the selected stakeholder), repeat.
 
 ### Listeners
 
-Listeners handle incoming messages and respond to them. Various
-supplemental listeners will not be covered, focusing on the main ones
-instead:
+Listeners handle incoming messages and respond to them. 
+Various supplemental listeners will not be covered, focusing on the main ones instead.
+
+Mostly listeners use *Relay* framework, which inludes three type of messages:
```
Comment (ChShersh) [inline _docs/for-contributors/2017-01-02-implementation.md]: Maybe make sense to add link to high level description of what is *Relay*: _Inv/Req/Data and MessagePart_ in https://cardano-docs.iohk.io/protocols/csl-application-level/

Comment (pva701) [inline _docs/for-contributors/2017-01-02-implementation.md]: done

**Pull request merged**